File: Selenium/Activities/Activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver import ActionChains

with webdriver.Firefox() as driver:
    Actions = webdriver.ActionChains(driver)
    
    driver.get("https://training-support.net/webelements/mouse-events")
    ##Get the title of the page and print it to the console
    print("Title of the page: ", driver.title)

    ##Find the elements
    cargolock = driver.find_element(By.XPATH, "//h1[text()='Cargo.lock']")
    cargotoml = driver.find_element(By.XPATH, "//h1[text()='Cargo.toml']")
    src = driver.find_element(By.XPATH, "//h1[text()='src']")
    target = driver.find_element(By.XPATH, "//h1[text()='target']")

    ##Left click on the Cargo.lock button, move the cursor to the Cargo.toml button and then click it
    Actions.click(cargolock).pause(1).move_to_element(cargotoml).pause(5).click(cargotoml).perform()
    
    ##Print the confirmation text
    actionMessage = driver.find_element(By.ID, "result").text
    print(actionMessage)
    
    ##Double click on the src button. Then right click on the target button and select open
    Actions.double_click(src).pause(2).move_to_element(target).pause(5).context_click(target).perform()
    Actions.click(driver.find_element(By.XPATH, ("//div[@id='menu']/div/ul/li[1]"))).pause(5).perform()
    
    ##Print the confirmation text
    actionMessage=driver.find_element(By.ID, "result").text
    print(actionMessage)
    
    ##Close the browser
    driver.quit()

File: Selenium/Activities/Activity8.java
package Activity;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity8 {
	
	public static void main(String[] args) {
		
		WebDriver driver = new FirefoxDriver();
		
		Actions builder = new Actions(driver);
		
		driver.get("https://training-support.net/webelements/mouse-events");
		System.out.println("Title of the page: " + driver.getTitle());
		
		//find the elements
		WebElement cargolock = driver.findElement(By.xpath("//h1[text()='Cargo.lock']"));
		WebElement cargotoml = driver.findElement(By.xpath("//h1[text()='Cargo.toml']"));
		WebElement src = driver.findElement(By.xpath("//h1[text()='src']"));
		WebElement targetButton = driver.findElement(By.xpath("//h1[text()='target']"));
		
		builder.click(cargolock).pause(1000).moveToElement(cargotoml).pause(500).click(cargotoml).build().perform();
		
		String actionMessage = driver.findElement(By.id("result")).getText();
		
		System.out.println("actionMessage");
		
		
		builder.doubleClick(src).pause(5000).contextClick(targetButton).pause(3000).build().perform();
		
		builder.click(driver.findElement(By.xpath("//div[@id='menu']/div/ul/li[1]"))).pause(5000).build().perform();
        // Print the front side text
        actionMessage = driver.findElement(By.id("result")).getText();
        System.out.println(actionMessage);

        // Close the browser
        driver.quit();
			
	}
	
}
